### PR TITLE
Propagator evolution. Change in interface.

### PR DESCRIPTION
**Pull request by amitjamadagni**
`QuPropagator` is mapped to `QuStateEvolution`. `QuPropagatorState` is used for both `QuStateEvolution` and `QuPropagatorEvolution`. This aims to address the issue #29 .

### CODE REVIEW

**Comment by amitjamadagni** [comment]
@acroy thoughts on this would be helpful. Thanks !

**Comment by acroy** [comment]
I think the change to `QuStateEvolution` makes sense. I am not sure if we want to have the iterator interface also for the evolution operator. I think one really wants `QuPropagator(H,tau)` (or `QuPropagator(H,te,ti)`) to be a `QuMatrix`.

File: src/propmachinery.jl [hunk 1]
```diff
@@ -140,7 +140,16 @@ Inputs :
 Output :
 * True if the current state is final state, else false.
 """ ->
-Base.done(prob::QuPropagator, qustate::QuPropagatorState) = done(prob.tlist, qustate.t_state)
+Base.done(prob::QuStateEvolution, qustate::QuPropagatorState) = done(prob.tlist, qustate.t_state)
 
-export  QuPropagator,
+QuPropagator{QM<:QuBase.AbstractQuMatrix}(op::QM, dt::Float64) = expm(-im*op*dt)
```
Comment (acroy) [inline src/propmachinery.jl]: Here we should add some comment/docs on what we are trying to do.

File: test/propagatortest.jl [hunk 2]
```diff
@@ -127,3 +127,8 @@ end
 
 ind = findin(tlist, rand(tlist))[1]
 @test_approx_eq_eps coeffs(vec(rhos[ind])) coeffs(vec(ps[ind])) 1e-1
+
+# tests for  evolution opeator..
```
Comment (acroy) [inline test/propagatortest.jl]: I think it is better to apply the QuPropagator to an initial state and compare that to a state propagated with some other method (maybe Expokit).

**Comment by acroy** [comment]
Although it makes sense to rename QuPropagator, maybe it is not a good idea at this point since in all blogs etc we have called it QuPropagator. Of course we don't have an official release yet, but still. We could do the following: keep `QuStateEvolution` as you have it now, `typedef QuPropagator QuStateEvolution` and use `QuPropagatorOp` or `QuEvolutionOp` for the time-evolution operator?

**Comment by amitjamadagni** [comment]
@acroy I guess this is ready !

**Comment by acroy** [comment]
IMO `QuOperatorEvolution` is not so appropriate, as it implies somehow that it propagates operators, which it doesn't.

Also all the tests for `QuOperatorEvolution` should be done in the same way (i.e. applying to an initial state and comparing to some other propagation method).

**Comment by amitjamadagni** [comment]
@acroy a review would be helpful !

**Comment by acroy** [comment]
Looks good so far. Did you check if the examples are still running?

**Comment by amitjamadagni** [comment]
@acroy yup ! Though there is a lag on the example on mcwf (checked this with reduced number of trajectories).

**Pull request merged**